internal/controller/management/projectconfigs/project_configs.go

```diff
@@ -0,0 +1,311 @@
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionFalse,
+		Reason:             "Syncing",
+		Message:            "Ensuring project config webhook receivers",
+		ObservedGeneration: pc.GetGeneration(),
+	})
+
+	whReceivers, err := r.syncWebhookReceiversFn(ctx, pc)
+	status.WebhookReceivers = whReceivers
+	if err != nil {
+		logger.Error(err, "error syncing webhook receivers",
+			"project-config", pc.Name,
+		)
+		conditions.Set(status, &metav1.Condition{
+			Type:               kargoapi.ConditionTypeReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "SecretMisconfiguration",
+			Message:            "Failed to sync webhook receivers: " + err.Error(),
+			ObservedGeneration: pc.GetGeneration(),
+		})
+		return *status, true, err
+	}
+
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Synced",
+		Message:            "ProjectConfig is synced and ready for use",
+		ObservedGeneration: pc.GetGeneration(),
+	})
+	conditions.Delete(status, kargoapi.ConditionTypeReconciling)
+	return *status, false, nil
+}
+
+func (r *reconciler) syncWebhookReceivers(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) ([]kargoapi.WebhookReceiver, error) {
+	logger := logging.LoggerFromContext(ctx)
+	if pc.Spec.WebhookReceivers == nil {
+		logger.Debug("ProjectConfig does not have any receiver configurations")
+		return nil, nil
+	}
+
+	logger.Debug("syncing webhook receivers",
+		"webhook-receiver-configs", len(pc.Spec.WebhookReceivers),
+	)
+
+	var err error
+	var webhookReceivers []kargoapi.WebhookReceiver
+	for _, rc := range pc.Spec.WebhookReceivers {
+		whr, initErr := r.newWebhookReceiver(ctx, pc, rc)
+		if initErr != nil {
+			logger.Error(err, "error initializing new webhook receiver",
+				"receiver-config", rc,
+			)
+			err = multierr.Append(initErr,
+				fmt.Errorf("error initializing webhook receiver %q: %w",
+					rc.Name, initErr,
+				),
+			)
+			continue
+		}
+		webhookReceivers = append(webhookReceivers, *whr)
+	}
+	return webhookReceivers, err
+}
+
+func (r *reconciler) newWebhookReceiver(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+	rc kargoapi.WebhookReceiverConfig,
+) (*kargoapi.WebhookReceiver, error) {
+	logger := logging.LoggerFromContext(ctx)
+	cfg, err := getProviderConfig(rc)
+	if err != nil {
+		logger.Error(err, "error getting provider config",
+			"receiver-config", rc,
+			"project-config", pc,
+		)
+		return nil, fmt.Errorf("error getting secret name: %w", err)
+	}
+
+	var s corev1.Secret
+	err = r.client.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: pc.Namespace,
+			Name:      cfg.secretName,
+		},
+		&s,
+	)
+	if err != nil {
+		logger.Error(err, "secret not found",
+			"secret-name", cfg.secretName,
+		)
+		if kubeerr.IsNotFound(err) {
+			return nil, fmt.Errorf(
+				"secret-reference name %q in namespace %q not found",
```

Comment:
```suggestion
				"secret %q in namespace %q not found",
```

bears the same message.